src/main/java/com/dreamsportslabs/guardian/service/Passwordless.java

```diff
@@ -63,6 +63,11 @@ public Single<PasswordlessModel> init(
                 throw RESENDS_EXHAUSTED.getException();
               }
 
+              if (model.getTries() >= model.getMaxTries()) {
+                passwordlessDao.deletePasswordlessModel(model.getState(), tenantId);
+                throw RETRIES_EXHAUSTED.getException();
+              }
+
```

Comment:
Not required here, this check should be present in complete flow. tries are incremented in the complete flow.